ethers-solc/tests/project.rs

```diff
@@ -426,12 +429,31 @@ fn can_flatten_file_with_duplicates() {
     assert!(result.is_ok());
 
     let result = result.unwrap();
+    assert!(!result.contains("import"));
     assert_eq!(result.matches("contract Foo {").count(), 1);
     assert_eq!(result.matches("contract Bar {").count(), 1);
     assert_eq!(result.matches("contract FooBar {").count(), 1);
     assert_eq!(result.matches(';').count(), 1);
 }
 
+#[test]
+fn can_flatten_on_solang_failure() {
+    let root =
+        PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("test-data/test-flatten-solang-failure");
+    let paths = ProjectPathsConfig::builder().sources(&root.join("contracts"));
+    let project = TempProject::<ConfigurableArtifacts>::new(paths).unwrap();
+
+    let target = root.join("contracts/Contract.sol");
+
+    let result = project.flatten(&target);
+    assert!(result.is_ok());
+
+    let result = result.unwrap();
+    assert!(!result.contains("import"));
+    assert_eq!(result.matches("contract Contract {").count(), 1);
+    assert_eq!(result.matches("library Lib {").count(), 1);
```

Comment:
can we compare against the whole string instead, maybe even for the other test cases?

Reply:
yeah, we can do that